packages/openfdd-agent-shell/src/openfdd_agent_shell/cron/scheduler.py

```
@@ -0,0 +1,190 @@
+            text = str(payload.get("text") or payload.get("message") or job.name)
+            path = self.memory.remember(text)
+            return f"appended memory: {path}"
+        if job.service == "shell":
+            command = payload.get("command")
+            if not command:
+                raise ValueError("shell jobs require payload.command")
+            if isinstance(command, str):
+                command_list = command
+            else:
+                command_list = [str(part) for part in command]
+            proc = subprocess.run(
```

Ruff (0.15.14)

[error] `subprocess` call: check for execution of untrusted input

(S603)